common/Freemium.hpp

```diff
@@ -0,0 +1,34 @@
+/* -*- Mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4; fill-column: 100 -*- */
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef ONLINE_COMMON_FREEMIUM
+#define ONLINE_COMMON_FREEMIUM
+
+#include <string>
+#include <vector>
+
+namespace Freemium
+{
+class FreemiumManager
+{
+    static std::vector<std::string> FreemiumBlacklist;
+    static bool _isFreemiumUser;
+    static std::string FreemiumBlacklistString;
+
+    static void generateBlacklist();
+
+public:
+    FreemiumManager();
+    static const std::vector<std::string>& getFreemiumBlacklist();
+    static const std::string getFreemiumBlacklistString();
+
+    static bool& isFreemiumUser() { return _isFreemiumUser; }
```

Comment:
Why return a reference? Surely we don't want the caller to change this flag. 

`static bool isFreemiumUser() { return _isFreemiumUser; }` ?

Reply:
it's because of this [line](https://github.com/CollaboraOnline/online/pull/2440/files#diff-5d08e62594b9952da25eb814e4567987fc57740d3c665dc789b824e26c2da6bbR881) I can have a set method(not set with the parameter but returning a reference) maybe or a public variable or whatever workaround you suggest. But this was intentional here.

Reply:
it was intentional but I realised there could be another way. thanks